src/gcp_airflow_foundations/source_class/source.py

```diff
@@ -45,40 +69,107 @@ def build_dags(self):
             logging.info(f"table_default_task_args {table_default_task_args}")
 
             start_date = table_default_task_args["start_date"]
-
+            
+            kwargs = data_source.dag_args if data_source.dag_args else {}
+            
             with DAG(
-                dag_id=f"{data_source.name}_to_bq_{table_config.table_name}",
+                dag_id=f"{data_source.name}.{table_config.table_name}",
                 description=f"{data_source.name} to BigQuery load for {table_config.table_name}",
                 schedule_interval=data_source.ingest_schedule,
                 default_args=table_default_task_args,
-                render_template_as_native_obj=True
+                render_template_as_native_obj=True,
+                **kwargs
             ) as dag:    
 
-                load_to_bq_landing = self.get_bq_ingestion_task(dag, table_config)
+                self.set_task_dependencies(dag, table_config, data_source, "schema_parsing")
+                dags.append(dag)
+
+        extra_dags=  self.get_extra_dags()
+
+        if extra_dags is not None and not extra_dags == []:
+            dags = dags + self.get_extra_dags()
+
+        return dags
+
+    def build_dags_source_level(self):
```

Comment:
the code in the two build_dags methods can probably be consolidated into one in a cleaner fashion

Reply:
Yea, I may be over simplifying. But the way I imagined it is that there is one functions called to create an ingestion for a table and it accepts a dag and (optional)task_group as a param. 

If it is a DAG per table config, the taskgroup is NONE. And only thing that needs to change around the function is the for look for generating  DAG/TaskGroups